pkg/metrics/producers/factory.go

```diff
@@ -34,16 +34,16 @@ type Factory struct {
 }
 
 func (f *Factory) For(mp *v1alpha1.MetricsProducer) metrics.Producer {
-	if err := mp.Spec.Validate(); err != nil {
-		return &fake.FakeProducer{WantErr: err}
-	}
 	if mp.Spec.PendingCapacity != nil {
 		return &pendingcapacity.Producer{
 			MetricsProducer: mp,
 			Client:          f.Client,
 		}
 	}
 	if mp.Spec.Queue != nil {
+		if err := mp.Spec.ValidateQueue(); err != nil {
```

Comment:
You shouldn't need to validate here. Validation logic should have caught this such that it could never be invalid. 

Even if you assume the webhook isn't installed, we just run a cascading Validate as the first line of the reconciler. Perhaps this should live here: https://github.com/awslabs/karpenter/blob/main/pkg/controllers/controller.go#L65